java/fury-core/src/main/java/org/apache/fury/resolver/ClassResolver.java

```diff
@@ -1701,7 +1701,7 @@ public Class<?> xreadClass(MemoryBuffer buffer) {
     Class<?> cls = classNameBytes2Class.get(byteString);
     if (cls == null) {
       Preconditions.checkNotNull(byteString);
-      String className = byteString.decode('.', '_');
+      String className = byteString.decode(Encoders.PACKAGE_DECODER);
```

Comment:
Should not be PACKAGE_DECODER here for classname decode

Reply:
I saw that the [ClassInfo constructor](https://github.com/apache/incubator-fury/blob/main/java/fury-core/src/main/java/org/apache/fury/resolver/ClassInfo.java#L84-L85) also uses PACKAGE_ENCODER encoding `Class#getName()`

Reply:
Done